### DISCUSSION TITLE
[BUG] Bad Request: query is too old and response timeout expired or query ID is in

### DISCUSSION THREAD

**Discussion by darkvovich** [nutgram/nutgram#734] [Q&A]
### Nutgram version

4.2.1

### PHP version

8.2

### Framework

No Framework

### Cache driver

FileCache

### OS

Debian

### Describe the bug

In log files often see this error
```
PHP message: PHP Fatal error:  Uncaught GuzzleHttp\Exception\ClientException: Client error: `POST https://api.telegram.org/bot[REDACTED]/answerCallbackQuery` resulted in a `400 Bad Request` response:
{"ok":false,"error_code":400,"description":"Bad Request: query is too old and response timeout expired or query ID is in (truncated...)
 in /vendor/guzzlehttp/guzzle/src/Exception/RequestException.php:113
Stack trace:
#0 /vendor/guzzlehttp/guzzle/src/Middleware.php(72): GuzzleHttp\Exception\RequestException::create()
#1 /vendor/guzzlehttp/promises/src/Promise.php(209): GuzzleHttp\Middleware::GuzzleHttp\{closure}()
#2 /vendor/guzzlehttp/promises/src/Promise.php(158): GuzzleHttp\Promise\Promise::callHandler()
#3 /vendor/guzzlehttp/promises/src/TaskQueue.php(52): Guzz"
```

### To Reproduce

1. 

### Expected behavior

How to detect and fix it?

**Comment by Lukasss93** [comment]
Hi!

You need to responde with `$bot->answerCallbackQuery();` as soon as possible when a user interact with an inline keyboard.
https://core.telegram.org/bots/api#answercallbackquery

CallbackQuery (and InlineQuery for that matter) generally have an answer timeframe of 10s, if you take any longer to respond, the query id becomes invalid and you get the error
I suggest you time your bot's server to check how long does it take to process an incoming query, if it takes close to or more than 10s, you know to optimize the code that takes too long to process.

How to handle Telegram errors:
https://nutgram.dev/docs/usage/handlers#onapierror

Anyway, here a way to detect and mute this error:

1. Create an exception class:
```php
<?php
// QueryTooOldException.php
namespace App\Telegram\Exceptions;

use SergiX44\Nutgram\Exception\ApiException;
use SergiX44\Nutgram\Nutgram;
use SergiX44\Nutgram\Telegram\Exceptions\TelegramException;

class QueryTooOldException extends ApiException
{
	// capture the "query is too old" error
    public static ?string $pattern = '.*query is too old.*';

    public function __invoke(Nutgram $bot, TelegramException $e)
    {
        // override this method with an empty content to ignore this exception
		// ignore me :P
    }
}
``` 
2. Register the exception
```php
$bot->registerApiException(QueryTooOldException::class);
``` 

PS: Remember to update Nutgram to the latest version 😉

**Reply by darkvovich** [reply]
Thx!
What didderence between $bot->answerCallbackQuery(); and $this->end(); in conversation?

**Reply by Lukasss93** [reply]
The "answerCallbackQuery" method tells Telegram that the bot handled the button click.
The "end" method ends a Nutgram conversation.

Look at this example: https://github.com/Lukasss93/telegram-stickeroptimizer/blob/master/app/Telegram/Conversations/FeedbackConversation.php
